include/taco/lower/lowerer_impl.h

```diff
@@ -54,24 +54,24 @@ class LowererImpl : public util::Uncopyable {
   virtual ~LowererImpl() = default;
 
   /// Lower an index statement to an IR function.
-  ir::Stmt lower(IndexStmt stmt, std::string name, 
-                 bool assemble, bool compute, bool pack, bool unpack);
+  virtual ir::Stmt lower(IndexStmt stmt, std::string name, 
+                 bool assemble, bool compute, bool pack, bool unpack) = 0;
 
 protected:
 
   /// Lower an assignment statement.
-  virtual ir::Stmt lowerAssignment(Assignment assignment);
+  virtual ir::Stmt lowerAssignment(Assignment assignment) = 0;
 
   /// Lower a yield statement.
-  virtual ir::Stmt lowerYield(Yield yield);
+  virtual ir::Stmt lowerYield(Yield yield) = 0;
 
 
   /// Lower a forall statement.
-  virtual ir::Stmt lowerForall(Forall forall);
+  virtual ir::Stmt lowerForall(Forall forall) = 0;
 
   /// Lower a forall that needs to be cloned so that one copy does not have guards
   /// used for vectorized and unrolled loops
-  virtual ir::Stmt lowerForallCloned(Forall forall);
+  virtual ir::Stmt lowerForallCloned(Forall forall) = 0;
```

Comment:
Do the helper methods for `lowerForall` (this method, `searchForFusedPositionStart`, `resolveCoordinate`, etc.) really need to be virtual methods of the base lowerer? Would they actually be needed for every target? (For instance, I'd thought `resolveCoordinate` might not be needed for architectures like ExTensor that have hardware support for intersections/unions.)

Reply:
You're right, I don't think they'd be needed for every target but I just made every method in the original LowererImpl an abstract method. I can make this a little cleaner today and let you know when they've been removed from the LowererImpl Abstract Class

Reply:
Resolved 777519d